cartography/intel/kubernetes/services.py

```diff
@@ -0,0 +1,85 @@
+import logging
+from typing import Dict
+from typing import List
+
+from neo4j import Session
+
+from cartography.intel.kubernetes.util import get_epoch
+from cartography.intel.kubernetes.util import K8sClient
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+@timeit
+def sync_services(
+    session: Session, client: K8sClient, update_tag: int, cluster: Dict, pods: List[Dict],
+) -> None:
+    services = list()
+    for service in client.core.list_service_for_all_namespaces().items:
+        item = {
+            "uid": service.metadata.uid,
+            "name": service.metadata.name,
+            "creation_timestamp": get_epoch(service.metadata.creation_timestamp),
+            "deletion_timestamp": get_epoch(service.metadata.deletion_timestamp),
+            "namespace": service.metadata.namespace,
+            "cluster_uid": cluster["uid"],
+            "type": service.spec.type,
+            "selector": service.spec.selector,
+            "load_balancer_ip": service.spec.load_balancer_ip,
+        }
+
+        ingresses = service.status.load_balancer.ingress
+        for ingress in ingresses or list():
+            item.update({"ingress_host": ingress.hostname, "ingress_ip": ingress.ip})
+
+        service_pods = list()
+        for pod in pods:
+            is_service_pod = True if service.spec.selector else False
+            for selector in service.spec.selector or dict():
+                if (
+                    selector not in pod["labels"] or
+                    service.spec.selector[selector] != pod["labels"][selector]
+                ):
+                    is_service_pod = False
+                    break
+            if is_service_pod:
+                service_pods.append(pod)
+        item["pods"] = service_pods
+        services.append(item)
+    load_service_data(session, services, update_tag)
+
+
+def load_service_data(session: Session, data: List[Dict], update_tag: int) -> None:
+    ingestion_cypher_query = """
+    UNWIND {services} as k8service
+        MERGE (service:KubernetesService {id: k8service.uid})
+        ON CREATE SET service.firstseen = timestamp()
+        SET service.lastupdated = {update_tag},
+            service.name = k8service.name,
+            service.created_at = k8service.creation_timestamp,
+            service.deleted_at = k8service.deletion_timestamp,
+            service.type = k8service.type,
+            service.load_balancer_ip = k8service.load_balancer_ip,
+            service.ingress_host = k8service.ingress_host,
+            service.ingress_ip = k8service.ingress_ip
+        WITH service, k8service.namespace as ns, k8service.cluster_uid as cuid, k8service.pods as k8pods
+        MATCH (space:KubernetesNamespace {name: ns})-[:IN_CLUSTER]->(cluster:KubernetesCluster {id: cuid})
+        MERGE (space)-[rel1:HAS_SERVICE]->(service)
```

Comment:
I don't think you need to specify this in every file. Once this relationship is mapped, mapping it again just make the code redundant. Correct me if I'm wrong?

Reply:
`MATCH` here is used to get the namespace that belongs to the correct cluster; otherwise, we may end up mapping things to a namespace from wrong cluster because the name of namespace can be same across different clusters.

Reply:
I see. @achantavy Could you please give this a quick look?